core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -872,6 +917,387 @@ export class ExplorationEditor extends BaseUser {
+   */
+  async navigateToHistoryTab(): Promise<void> {
+    await this.navigateToTab(mobileHistoryTabButton, HistoryTabButton);
+  }
+
+  /**
+   * Checks if a specific revision has certain properties.
+   * @param {number} revisionNumber - The number of the revision to check.
+   * @param {string[]} properties - The properties to check for in the revision.
+   */
+
+  async expectRevisionToHave(
+    revisionNumber: number,
+    properties: string[]
+  ): Promise<void> {
+    if (!Number.isInteger(revisionNumber) || revisionNumber <= 0) {
+      throw new Error('Invalid revision number');
+    }
+
+    let elements = await this.page.$$(historyListItem);
+    if (elements.length < revisionNumber) {
+      throw new Error(
+        `There are not enough revisions. Requested: ${revisionNumber}, available: ${elements.length}`
+      );
+    }
+
+    let element = elements[revisionNumber - 1];
+
+    const propertyToSelector: {[key: string]: string} = {
+      'Version No.': revisionVersionNoSelector,
+      Notes: revisionNoteSelector,
+      User: revisionUsernameSelector,
+      Date: revisionDateSelector,
+    };
+
+    for (let property of properties) {
+      let selector = propertyToSelector[property];
+      if (!selector) {
+        throw new Error(`Invalid property: ${property}`);
+      }
+
+      let value = await element.$eval(selector, async el => el.textContent);
+      if (property !== 'Notes' && (!value || value.trim() === '')) {
+        throw new Error(
+          `Revision ${revisionNumber} is missing or has an empty ${property}`
+        );
+      }
+
+      showMessage(`Revision ${revisionNumber} has the property ${property}`);
+    }
+  }
+
+  /**
+   * Checks if the revisions are ordered by a specific property.
+   * @param {string} property - The property to check the order by.
+   */
+  async expectRevisionsToBeOrderedBy(property: string): Promise<void> {
+    if (!property) {
+      throw new Error('Invalid property');
+    }
+
+    let revisions = await this.page.$$(historyListItem);
+    for (let i = 0; i < revisions.length - 1; i++) {
+      let value1 = await revisions[i].$eval(property, el => el.textContent);
+      let value2 = await revisions[i + 1].$eval(property, el => el.textContent);
+
+      if (!value1 || !value2) {
+        throw new Error(`Failed to retrieve property ${property}`);
+      }
+
+      if (typeof value1 === 'string') {
+        value1 = value1.toString();
+        value2 = value2.toString();
+      }
+
+      if (value1 < value2) {
+        throw new Error(`Revisions are not sorted by ${property}`);
+      }
+    }
+
+    showMessage(`Revisions are sorted by ${property}`);
+  }
+
+  /**
+   * Filters the revisions by a specific user.
+   * @param {string} userName - The name of the user to filter the revisions by.
+
+   */
+  async filterRevisionsByUser(userName: string): Promise<void> {
+    if (!userName) {
+      throw new Error('Invalid user name');
```

Comment:
I think the error message should say `Username cannot be empty`. I can't imagine TS compiling successfully while allowing `null` and `undefined` to be passed as a `string` argument. So that just leaves us with the empty string `''`.